### ISSUE TITLE
All Kotlin Spring Boot Producer with Gradle build system, Kotlin version incompatibility

### ISSUE THREAD

**Issue by nbrinks** [spring-cloud-samples/spring-cloud-contract-samples#115]
I am trying to do something similar to the sample `Producer Kotlin FTW` but keep running into an issue with the Kotlin version.
* **Gradle:** 6.2
* **Spring Boot:** 2.2.5.RELEASE
* **Spring Cloud:** Hoxton.SR3
* **Spring Cloud Contracts:** 2.2.2.RELEASE
* **Kotlin:** 1.3.70

I have multiple contracts written as `.kts`
Whenever I try upgrading the version on Kotlin on my project past `1.3.31` I can no longer generate the contract tests e.g. `./gradlew generateContractTests`.

<details><summary>Stacktrace</summary>

```
./gradlew generateContractTests --stacktrace
> Task :generateContractTests FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateContractTests'.
> org/jetbrains/kotlin/cli/jvm/repl/GenericReplCompiler

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':generateContractTests'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:205)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:203)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:184)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/cli/jvm/repl/GenericReplCompiler
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine$replCompiler$2.invoke(KotlinJsr223JvmLocalScriptEngine.kt:47)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine$replCompiler$2.invoke(KotlinJsr223JvmLocalScriptEngine.kt:38)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine.getReplCompiler(KotlinJsr223JvmLocalScriptEngine.kt)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine$localEvaluator$2.invoke(KotlinJsr223JvmLocalScriptEngine.kt:53)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine$localEvaluator$2.invoke(KotlinJsr223JvmLocalScriptEngine.kt:38)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine.getLocalEvaluator(KotlinJsr223JvmLocalScriptEngine.kt)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine.getReplEvaluator(KotlinJsr223JvmLocalScriptEngine.kt:55)
        at org.jetbrains.kotlin.script.jsr223.KotlinJsr223JvmLocalScriptEngine.createState(KotlinJsr223JvmLocalScriptEngine.kt:59)
        at org.jetbrains.kotlin.cli.common.repl.KotlinJsr223JvmScriptEngineBase.createState$default(KotlinJsr223JvmScriptEngineBase.kt:46)
        at org.jetbrains.kotlin.cli.common.repl.KotlinJsr223JvmScriptEngineBase.getCurrentState(KotlinJsr223JvmScriptEngineBase.kt:53)
        at org.jetbrains.kotlin.cli.common.repl.KotlinJsr223JvmScriptEngineBase.nextCodeLine(KotlinJsr223JvmScriptEngineBase.kt:44)
        at org.jetbrains.kotlin.cli.common.repl.KotlinJsr223JvmScriptEngineBase.compileAndEval(KotlinJsr223JvmScriptEngineBase.kt:61)
        at org.jetbrains.kotlin.cli.common.repl.KotlinJsr223JvmScriptEngineBase.eval(KotlinJsr223JvmScriptEngineBase.kt:33)
        at org.springframework.cloud.contract.spec.internal.KotlinContractConverter$convertFrom$eval$1.invoke(KotlinContractConverter.kt:50)
        at org.springframework.cloud.contract.spec.internal.KotlinContractConverter$convertFrom$eval$1.invoke(KotlinContractConverter.kt:31)
        at org.springframework.cloud.contract.spec.internal.KotlinContractConverter.withUpdatedClassloader(KotlinContractConverter.kt:68)
        at org.springframework.cloud.contract.spec.internal.KotlinContractConverter.convertFrom(KotlinContractConverter.kt:46)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.tryConvert(ContractFileScanner.groovy:308)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.addContractToTestGeneration(ContractFileScanner.groovy:286)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.appendRecursively(ContractFileScanner.groovy:253)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.addContractToTestGeneration(ContractFileScanner.groovy:295)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.appendRecursively(ContractFileScanner.groovy:253)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.addContractToTestGeneration(ContractFileScanner.groovy:295)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.appendRecursively(ContractFileScanner.groovy:253)
        at org.springframework.cloud.contract.verifier.file.ContractFileScanner.findContractsRecursively(ContractFileScanner.groovy:221)
        at org.springframework.cloud.contract.verifier.TestGenerator.generateTestClasses(TestGenerator.groovy:125)
        at org.springframework.cloud.contract.verifier.TestGenerator.generate(TestGenerator.groovy:104)
        at org.springframework.cloud.contract.verifier.plugin.GenerateServerTestsTask.generate(GenerateServerTestsTask.groovy:132)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:727)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:694)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:568)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:553)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:536)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:276)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:265)
        at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
        at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
        at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:153)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:67)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:41)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:192)
        ... 30 more
```

</details>

I noticed in the sample the following code.

https://github.com/spring-cloud-samples/spring-cloud-contract-samples/blob/a2d27e77796d3b2963a33b91b06d6b8eeb763782/producer_kotlin_ftw/build.gradle.kts#L28-L30

Can someone point me in the right direction because I don't entirely understand the problem here?
Why does Gradle need to be aligned with the version of Kotlin? What needs to change to allow my project to use a newer version of Kotlin?

**Comment by marcingrzejszczak** [comment]
Gradle has a provided version of Kotlin. Boot has a different version of Kotlin and we're reusing Boot's version of Kotlin so AFAIR you have to set the version of Kotlin to be aligned with Gradle

**Comment by nbrinks** [comment]
Thanks for the additional information @marcingrzejszczak 

> Gradle has a provided version of Kotlin.
I guess I didn't realize Gradle would include its own version of Kotlin.

I think there might be a bug though, or it's not as simple as aligning the versions.
```
./gradlew --version

------------------------------------------------------------
Gradle 6.2
------------------------------------------------------------

Build time:   2020-02-17 08:32:01 UTC
Revision:     61d3320259a1a0d31519bf208eb13741679a742f

Kotlin:       1.3.61
Groovy:       2.5.8
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.6 (Azul Systems, Inc. 11.0.6+10-LTS)
OS:           Mac OS X 10.15.3 x86_64
```
According to that output, Gradle is using Kotlin `1.3.61`. However, setting my Kotlin version to `1.3.61` still causes the same problem.
```
    kotlin("jvm") version "1.3.61"
    kotlin("plugin.spring") version "1.3.61"
```
I noticed:

Kotlin version 1.3.70:
https://github.com/JetBrains/kotlin/blob/v1.3.70/plugins/scripting/scripting-compiler/src/org/jetbrains/kotlin/scripting/compiler/plugin/repl/GenericReplCompiler.kt

Kotlin version 1.3.61
https://github.com/JetBrains/kotlin/blob/v1.3.61/plugins/scripting/scripting-compiler/src/org/jetbrains/kotlin/scripting/compiler/plugin/repl/GenericReplCompiler.kt

Kotlin version 1.3.31 (Totally different package)
https://github.com/JetBrains/kotlin/blob/v1.3.31/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/repl/GenericReplCompiler.kt

Still digging into what is going on.

**Comment by marcingrzejszczak** [comment]
Thanks a lot, maybe there's a bug indeed :/

**Comment by marcingrzejszczak** [comment]
Actually we're not reusing Boot's version but we have a fixed one. I'm bumping it as we speak in the master branch for Gradle 6.2.2.

**Comment by nbrinks** [comment]
Interesting, where is that configured in Gradle? I poked around the source code but wasn't able to locate that type of thing.

I was going to say also, my `buildEnvironment` suggests it's trying to use `1.3.61`
```
+--- org.springframework.cloud:spring-cloud-contract-spec-kotlin:2.2.2.RELEASE
|    +--- org.springframework.cloud:spring-cloud-contract-spec-java:2.2.2.RELEASE
|    |    +--- org.codehaus.groovy:groovy:2.5.9
|    |    +--- org.springframework:spring-core:5.2.4.RELEASE
|    |    |    \--- org.springframework:spring-jcl:5.2.4.RELEASE
|    |    +--- dk.brics.automaton:automaton:1.11-8
|    |    +--- org.apache.commons:commons-text:1.7
|    |    |    \--- org.apache.commons:commons-lang3:3.9
|    |    \--- org.springframework:spring-jcl:5.2.4.RELEASE
|    +--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.31 -> 1.3.61
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.3.61
|    |    |    \--- org.jetbrains:annotations:13.0
|    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.61
|    |         \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    +--- org.jetbrains.kotlin:kotlin-compiler-embeddable:1.3.31 -> 1.3.61
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-script-runtime:1.3.61
|    |    +--- org.jetbrains.kotlin:kotlin-reflect:1.3.61
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-daemon-embeddable:1.3.61
|    |    \--- org.jetbrains.intellij.deps:trove4j:1.0.20181211
|    +--- org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:1.3.31 -> 1.3.61
|    |    +--- org.jetbrains.kotlin:kotlin-scripting-compiler-impl-embeddable:1.3.61
|    |    |    +--- org.jetbrains.kotlin:kotlin-scripting-common:1.3.61
|    |    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    |    |    |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.2.1
|    |    |    +--- org.jetbrains.kotlin:kotlin-scripting-jvm:1.3.61
|    |    |    |    +--- org.jetbrains.kotlin:kotlin-script-runtime:1.3.61
|    |    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    |    |    |    \--- org.jetbrains.kotlin:kotlin-scripting-common:1.3.61 (*)
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    |    |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.2.1
|    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.61 (*)
|    \--- org.jetbrains.kotlin:kotlin-script-util:1.3.31
|         +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.31 -> 1.3.61 (*)
|         +--- org.jetbrains.kotlin:kotlin-script-runtime:1.3.31 -> 1.3.61
|         +--- org.jetbrains.kotlin:kotlin-scripting-jvm:1.3.31 -> 1.3.61 (*)
|         +--- org.jetbrains.intellij.deps:trove4j:1.0.20181211
|         \--- org.jetbrains.kotlin:kotlin-daemon-client:1.3.31 -> 1.3.61
|              +--- org.jetbrains.kotlin:kotlin-reflect:1.3.61 (*)
|              \--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.2.1
```

However, in IntelliJ it shows it trying to use `1.3.31`
![image](https://user-images.githubusercontent.com/1877109/77080507-5a24f000-69c7-11ea-81c2-47d1d3dbd8ff.png)

Thanks for looking at this!

**Comment by marcingrzejszczak** [comment]
It's in the main pom - https://github.com/spring-cloud/spring-cloud-contract/commit/f7709dac9cc80af91f6a9629dfdd2eb4aa6dbc09 this is the commit where I bump Kotlin to 1.3.61. The version will always come from Gradle however it appears that what doesn't come from Gradle comes from us, which in this case is a wrong version

**Comment by nbrinks** [comment]
Ah, ok, I see what's happening!
Temporarily I was able to get around this by excluding the `org.jetbrains.kotlin` group and importing the newer versions.

For example (where `kotinVersion` = 1.3.61),
```kotlin
buildscript {
    val springCloudContractVersion: String by project
    val kotlinVersion: String by project

    repositories {
        mavenCentral()
    }
    dependencies {
        classpath("org.springframework.cloud:spring-cloud-contract-spec-kotlin:$springCloudContractVersion") {
            exclude("org.jetbrains.kotlin")
        }

        classpath("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
        classpath("org.jetbrains.kotlin:kotlin-compiler-embeddable:$kotlinVersion")
        classpath("org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:$kotlinVersion")
        classpath("org.jetbrains.kotlin:kotlin-script-util:$kotlinVersion")
        classpath("org.jetbrains.kotlin:kotlin-daemon-embeddable:$kotlinVersion")
    }
```

Thanks again @marcingrzejszczak!

**Comment by shanman190** [comment]
@marcingrzejszczak, I think with the latest snapshots of Spring Cloud Contract this should now also be resolved.

For @nbrinks knowledge, we've now made some changes to the Spring Cloud Contract Gradle Plugin such that it no longer has conflicting Kotlin or Groovy libraries on the Gradle buildscript classpath. These changes should be available in the 2020.0.0 (codename: Illford) release coming soon!

**Comment by marcingrzejszczak** [comment]
Fixed via https://github.com/spring-cloud-samples/spring-cloud-contract-samples/commit/925af992636c9363d0b5200d73ad15408c16d84e
